addons/website/models/ir_module_module.py

```diff
@@ -382,7 +382,7 @@ def update_list(self):
     def update_theme_images(self):
         IrAttachment = self.env['ir.attachment']
         existing_urls = IrAttachment.search_read([['res_model', '=', self._name], ['type', '=', 'url']], ['url'])
-        existing_urls = [url_wrapped['url'] for url_wrapped in existing_urls]
+        existing_urls = set(map(lambda attach: attach['url'], existing_urls))
```

Comment:
```suggestion
        existing_urls = {url_wrapped['url'] for url_wrapped in existing_urls}
```
if goal is just to deduplicate?

Reply:
not really to deduplicate, goal was just to transform list of dict {'id':x, 'url': 'xxx'} to list of url.
And since we use it later just to see the presence of a url, I supposed that set of 100th of string was more efficient than list.
I'm going/doing wrong ?

Reply:
Nah reasoning seems correct. I'd still just replace the list comprehension with a set comprehension, that's what it's for.